st2common/st2common/bootstrap/actionsregistrar.py

```diff
@@ -124,7 +126,24 @@ def _register_action(self, pack, action):
                             (pack, pack_field))
 
         action_api = ActionAPI(**content)
-        action_api.validate()
+
+        try:
+            action_api.validate()
+        except jsonschema.ValidationError as e:
+            # We throw a more user-friendly exception on invalid parameter name
+            msg = str(e)
+
+            is_invalid_parameter_name = 'Additional properties are not allowed' in msg
+            is_invalid_parameter_name &= 'in schema[\'properties\'][\'parameters\']' in msg
+
+            if is_invalid_parameter_name:
+                parameter_name = re.search('\'(.+?)\' was unexpected', msg).groups()[0]
```

Comment:
At some point I will also look into upstream repo, probably shouldn't be too hard (famous last words) to make a change to include offending attribute on the exception class instance.